### PR TITLE
#161114862 A user should be able to favourite an article

### PR DESCRIPTION
**Pull request by Victor-Ugwueze**
#### What does this PR do?
- Favourite an article
   
#### Description of Task to be completed?
  - add following methods   to ArticleController   
      - addFavourite   
     - removeFavourite 
     - getAllFavourite

 - add endpoints
    - list all all favourite for user
   ````GET /api/v1/articles/favourites````

   - add article to favourite
   ```` POST /api/v1/articles/:articleId/favourite````

   - remove article from favourite
   ```` DELETE /api/v1/articles/:articleId/favourite````

#### How should this be manually tested?
    - Clone the repo
    - cd into the 'lotus-ah'
    - Run 'npm install'
    - Use the .env.sample format to create a .env file
    - Run 'npm run migrate'
    - Run 'npm run start:dev'

#### Any background context you want to provide?
  It's needed to complete [this feature](https://www.pivotaltracker.com/story/show/160451607 "send notification feature"). As a user should receive updates on his favourited articles.

#### What are the relevant pivotal tracker stories?
[#161114862](https://www.pivotaltracker.com/story/show/160451597 "reset password")
 
### Question
 NA

### Screenshot
 NA

### CODE REVIEW

File: server/controllers/ArticleController.js [hunk 1]
```diff
@@ -272,5 +274,115 @@ class ArticleController {
...
+  static getAllFavorite(req, res) {
+    const userId = req.decoded.id;
+    FavoriteArticle.findAll({
+      include: [
+        {
+          model: Article,
+        },
+      ],
+      where: { userId, }
+    }).then(result => res.status(200).json({ status: 'Success', result }))
```
Comment (KaiserPhemi) [inline server/controllers/ArticleController.js]: #### Can we not make this a one-liner please?
This should read:
```js
.then(result => {
  res.status(200)
      .json({
        status: 'Success',
        result
      })
})
```
This also apply to the next line of code and every where in the code base that this was implemented.

**Review by KaiserPhemi — CHANGES_REQUESTED** [review]

**Comment by Victor-Ugwueze** [comment]
I have implemented the feedback @KaiserPhemi. Thank you

**Review by obulaworld — APPROVED** [review]
LGTM

File: server/controllers/ArticleController.js (hunk 1 above)
Comment (KaiserPhemi) [reply server/controllers/ArticleController.js]: @Victor-Ugwueze 
You did not fix this??!!!

**Review by KaiserPhemi — CHANGES_REQUESTED** [review]

**Comment by KaiserPhemi** [comment]
@Victor-Ugwueze 
👎

**Comment by Victor-Ugwueze** [comment]
@KaiserPhemi, Please I am sorry. Can you look at the screenshot below

screen shot 2018-10-11 at 3 12 19 pm

**Comment by KaiserPhemi** [comment]
@Victor-Ugwueze 
Please fix merge conflicts.

**Review by KaiserPhemi — APPROVED** [review]

**Pull request merged**